src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py

```diff
@@ -272,6 +275,43 @@ def getDatasetLocations(self, datasets, account=None):
                                                              account=account)
             result[datasetPath] = self.cric.PNNstoPSNs(locations)
         return result
+    
+    def getDatasetLocationsFromMSPileup(self, datasetsWithDbsURL):
+        """
+        Returns a dictionary with the location of the datasets according to MSPileup
+        The definition of "location" here is a union of all sites holding at least
+        part of the dataset (defined by the DATASET grouping).
+        :param datasets: list of datasets
+        :param account: rucio account to use, if it is not provided we fallback to
+        pileup account for backward compatibility
+        :return: a dictionary of dataset locations, key'ed by the dataset name
+        """
+        
+        result = {}
+        for dbsUrl, datasets in datasetsWithDbsURL.items():
+            for dataset in datasets:
+                try:
+                    queryDict = {'query': {'pileupName': dataset},
+                                'filters': ['expectedRSEs', 'currentRSEs', 'pileupName', 'containerFraction', 'ruleIds']}
+                    pileUpinstance = '-testbed' if 'cmsweb-testbed' in dbsUrl else '-prod'
+                    msPileupUrl = f"https://cmsweb{pileUpinstance}.cern.ch/ms-pileup/data/pileup"
+                    self.logger.debug(f'dbsUrl: {dbsUrl}')
+                    self.logger.debug(f'msPileupUrl: {msPileupUrl}')
+                    doc = getPileupDocs(msPileupUrl, queryDict, method='POST')[0]
+                    currentRSEs = doc['currentRSEs']
+                    self.logger.debug(f'doc: {doc}')
+                    if len(currentRSEs) == 0:
+                        self.logger.warning(f'No RSE has a copy of the desired pileup dataset. Expected RSEs: {doc["expectedRSEs"]}')  
+                    self.logger.info(f'locationsFromPileup - name: {dataset}, currentRSEs: {doc["currentRSEs"]}, containerFraction: {doc["containerFraction"]}')
+                    result[dataset] = doc['currentRSEs']
+                except IndexError:
+                    self.logger.warning('Did not find any pileup document for query: %s', queryDict['query'])
+                    result[dataset] = []
+                    continue
```

Comment:
`continue` statement can actually be removed here, given that it will anyhow move to the next iteration.